crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -0,0 +1,243 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{
+    Arguments, CmpOp, Expr, ExprAttribute, ExprCall, ExprCompare, ExprContext, Identifier,
+};
+use ruff_python_semantic::Modules;
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+///
+/// Reports the following `re` calls when their first arguments are plain string
+/// literals, and no additional flags are passed:
+///
+/// - `sub`
+/// - `match`
+/// - `search`
+/// - `fullmatch`
+/// - `split`
+///
+/// For `sub`, the `repl` (replacement) argument must also be a string literal,
+/// not a function. For `match`, `search`, and `fullmatch`, the return value
+/// must also be used only for its truth value.
```

Comment:
docs nit: we prefer for this section to be a 1-2 sentence summary of the antipatterns the rule is looking for. These details are great, but should be put in a later section, in my opinion (maybe `## Details`, below the `## Why is this bad?` section?).

Here, I think the simple answer to "What it does?" is

```rs
/// Checks for uses of the `re` module that can be replaced with builtin `str` methods.